### ISSUE TITLE
New Rule: (no-duplicate-export) disallow export conflicts or overrides

### ISSUE THREAD

**Issue by spearmootz** [eslint/eslint#16657]
### Rule details

disallow of overwritting export varaiables

### Related ECMAScript feature

https://tc39.es/ecma262/multipage/ecmascript-language-scripts-and-modules.html#sec-exports

### What type of rule is this?

Warns about a potential problem

### Example code

```js
export const x = 'test';

// X will be missing
module.exports = 'test';
```

### Why should this rule be in the core instead of a plugin?

i think it is because it is related to language standards and is common enough when refactoring.

but if its better elsewhere let me know.

### Participation

- [ ] I am willing to submit a pull request to implement this rule.

### Additional comments

one way to accomplish this is to allow the following methods to coexist

(although these should fail because they are the same name, but pass otherwise)
export const varName
module.exports.varName

OR

module.exports = {}

but these two should not be mixed.

**Comment by mdjermanovic** [comment]
Hi @spearmootz, thanks for the issue!

> ```js
> export const x = 'test';
> 
> // X will be missing
> module.exports = 'test';
> ```

I'm not sure that I understand this request. `export const x = 'test';` is ESM export syntax that can be used only in ES modules while `module.exports = 'test';` is CommonJS export. A module cannot be ESM and CJS at the same time, so the problem here doesn't seem to be that `x` will be missing but that one of these two exports cannot be used depending on which kind of module the file is intended to be.

**Comment by spearmootz** [comment]
@mdjermanovic i am sorry, perhaps my example was mistaken, here is another.

```
module.exports.x = 'x'
module.exports = {y: 'y'};
```

right now there is no rule that prevents you from doing the example above.

**Comment by spearmootz** [comment]
which i guess then rule might be called something more like `no-export-overwrite`

**Comment by aladdin-add** [comment]
Per our [rules policies](https://eslint.org/blog/2020/05/changes-to-rules-policies), we can only accept new rules if they relate to new ECMAScript features that reached stage 4 in the preceding 12 months, so this proposal doesn't qualify.

In the meanwhile, you can check the plugin `eslint-plugin-n`'s rule `exports-style`: https://github.com/eslint-community/eslint-plugin-n/blob/master/docs/rules/exports-style.md
